lib/dune

```diff
@@ -12,6 +12,7 @@
   ocaml-version
   odoc_parser
   re
+  result
```

Comment:
I know it was introduced in #430, but I think we can remove dependency on this library altogether.

> Projects that want to use the new result type defined in OCaml >= 4.03 while staying compatible with older version of OCaml should use the Result module defined in this library.

Mdx works with `ocaml >= 4.08`...